### PR TITLE
Test using a pipeline created with null bind group layout

### PR DESCRIPTION
**Pull request by Jiawei-Shao**
This patch adds validation tests on the use of a render pipeline or a compute pipeline
created with null bind group layouts. Note that as the total amount of the sub cases
 (32) is not that large both `null` and `undefined` are tested in these tests.

Issue: #4075

<hr>

**Requirements for PR author:**

- [*] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [*] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [*] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [*] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by jzm-intel** [comment]
Since all cases are expected to success, is it possible to merge this new tests into the previous one `create_pipeline_with_null_bind_group_layouts`, such that "if the pipeline can be created successfully, it can be used in the render/compute pass encoder successfully"?

File: src/webgpu/api/validation/createPipelineLayout.spec.ts [hunk 1]
```diff
@@ -339,8 +367,9 @@ g.test('bind_group_layouts,create_pipeline_with_null_bind_group_layouts')
           code,
         });
 
+        const format = 'rgba8unorm';
         t.expectValidationError(() => {
```
Comment (jzm-intel) [inline src/webgpu/api/validation/createPipelineLayout.spec.ts:329]: How about rename line 349 `shouldError` -> `shouldCreatePipelineError`, and make the expectation like
``` typescript
        t.expectValidationError(() => {
          const renderPipeline = t.device.createRenderPipeline({
            layout,
            vertex: {
              module: shaderModule,
            },
            fragment: {
              module: shaderModule,
              targets: [
                {
                  format,
                },
              ],
            },
          });
        }, shouldCreatePipelineError);

        // If pipeline is created successfully, pass encoder can use it to test setBindGroup, 
        // which should always success.
        if (!shouldCreatePipelineError) {
          const renderTarget = t.createTextureTracked({
            usage: GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.COPY_SRC,
            size: [4, 4, 1],
            format,
          });
          const commandEncoder = t.device.createCommandEncoder();
          const renderPassEncoder = commandEncoder.beginRenderPass({
            colorAttachments: [
              {
                view: renderTarget.createView(),
                loadOp: 'load',
                storeOp: 'store',
              },
            ],
          });
          for (let i = 0; i < 4; ++i) {
            if (!setBindGroupOnEmptyBindGroupLayoutIndex && i === emptyBindGroupLayoutIndex) {
              continue;
            }
            renderPassEncoder.setBindGroup(i, bindGroups[i]);
          }
          renderPassEncoder.setPipeline(renderPipeline);
          renderPassEncoder.draw(1);
          renderPassEncoder.end();

          t.queue.submit([commandEncoder.finish()]);
        }

        break;
```

**Review by jzm-intel — COMMENTED** [review]
After second thought, we are testing two things if test cases merged (creating pipeline vs setting bind group), so it might be good to keep as separated tests (despite of some duplicated codes)?
@kainino0x What do you think? Sorry for troubling...

**Comment by kainino0x** [comment]
I haven't had a chance to actually review this (and I'm going on break - apologies).
But I don't think I have a strong opinion between the two options. Generally, since these are tests, I prefer duplicating some code over testing too many things at once (which can make it hard for reviewers/readers to determine whether the test is covering all of the necessary cases), but it's a balance and depends on what y'all think as reviewer and author.

**Comment by Jiawei-Shao** [comment]
> I haven't had a chance to actually review this (and I'm going on break - apologies). But I don't think I have a strong opinion between the two options. Generally, since these are tests, I prefer duplicating some code over testing too many things at once (which can make it hard for reviewers/readers to determine whether the test is covering all of the necessary cases), but it's a balance and depends on what y'all think as reviewer and author.

Done

**Comment by Jiawei-Shao** [comment]
The failure is not related to my PR.

PTAL, thanks!

**Comment by jzm-intel** [comment]
I proposed https://github.com/web-platform-tests/wpt/pull/49823 to fix the CI workflow.

**Review by jzm-intel — APPROVED** [review]
Looks good to me, thanks

**Comment by Jiawei-Shao** [comment]
Thanks a lot for your fix! @jzm-intel

**Pull request merged**